src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/Latin1Tests.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Server.IntegrationTesting.IIS;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests.InProcess
+{
+    [Collection(PublishedSitesCollection.Name)]
+    public class Latin1Tests : IISFunctionalTestBase
+    {
+        public Latin1Tests(PublishedSitesFixture fixture) : base(fixture)
+        {
+        }
+
+        [ConditionalFact]
+        [RequiresNewHandler]
+        public async Task Latin1Works()
+        {
+            var deploymentParameters = Fixture.GetBaseDeploymentParameters();
+            deploymentParameters.TransformArguments((a, _) => $"{a} AddLatin1");
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var client = new HttpClient(new LoggingHandler(new WinHttpHandler() { SendTimeout = TimeSpan.FromMinutes(3) }, deploymentResult.Logger));
+
+            var requestMessage = new HttpRequestMessage(HttpMethod.Get, $"{deploymentResult.ApplicationBaseUri}Latin1");
+            requestMessage.Headers.Add("foo", "£");
+
+            var result = await client.SendAsync(requestMessage);
+            Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+        }
+
+        [ConditionalFact]
+        [RequiresNewHandler]
+        public async Task Latin1ThrowsWithoutAppContextSwitch()
+        {
+            var deploymentParameters = Fixture.GetBaseDeploymentParameters();
+            deploymentParameters.TransformArguments((a, _) => $"{a}");
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var client = new HttpClient(new LoggingHandler(new WinHttpHandler() { SendTimeout = TimeSpan.FromMinutes(3) }, deploymentResult.Logger));
+
+            var requestMessage = new HttpRequestMessage(HttpMethod.Get, $"{deploymentResult.ApplicationBaseUri}Latin1");
+            requestMessage.Headers.Add("foo", "£");
+
+            var result = await client.SendAsync(requestMessage);
+            Assert.Equal(HttpStatusCode.InternalServerError, result.StatusCode);
```

Comment:
What actually happens on the server? The Assert.Equal fails because you get the ? replacement character? Write a test that checks for the ? then. This 500 response doesn't tell you anything about how the server handled it.

Reply:
How do you get the ? character lol.

Reply:
'\uFFFD' or something like that

Reply:
nvrm I got it I believe.